.github/workflows/ci.yaml

```diff
@@ -41,10 +41,15 @@ jobs:
         run: tox run -e unit
 
   build:
-    name: Build charms
-    uses: canonical/data-platform-workflows/.github/workflows/build_charms_with_cache.yaml@v5
+    strategy:
+      fail-fast: true
```

Comment:
interesting, what is this

Reply:
build_charms_with_cache and build_charms_without_cache are different workflows to build charms. One of the main differences between them is that `build_charms_with_cache` automatically discover ALL charms in current directory and its subdirectories, while build_charms_with_cache not. That leads to failures in tests because tests applications charms are not being built.

As I discussed with @carlcsaposs-canonical we implement a matrix that will run this workflow for all charms